src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1276,9 +1276,26 @@ public static Process Start(string fileName, IEnumerable<string> arguments)
                 throw new ArgumentNullException(nameof(arguments));
 
             var startInfo = new ProcessStartInfo(fileName);
-            foreach (string argument in arguments)
+            using (IEnumerator<string> e = arguments.GetEnumerator())
             {
-                startInfo.ArgumentList.Add(argument);
+                // Use Arguments if there's only one argument, or ArgumentList if there are multiple.
+                if (e.MoveNext())
+                {
+                    string firstArgument = e.Current;
+                    if (!e.MoveNext())
+                    {
+                        startInfo.Arguments = firstArgument;
```

Comment:
Are we still going to get the escaping of special characters with this optimization?

Reply:
You're right.  I forgot we made Argument and ArgumentList behave differently. It's unintuitive we did that,  but at least it's doc'd; even more so,  and not doc'd, that there's an escaping difference between the `Start(..., string)` and `Start(..., IEnumerable<string>)`.  And it's not worth complicating tons further to add escaping here.  Will undo.